### ISSUE TITLE
Error in CI: `Could not find a version that satisfies the requirement prelude-parser (from versions: none)`

### ISSUE THREAD

**Issue by apcamargo** [sanders41/python-project-generator#457]
The following line in `testing.yml` is causing an error in the CI pipeline:

```
python -m pip install --no-index --find-links=dist/ prelude-parser
```

I was unable to determine the purpose of this command, so I removed it for now. After doing so, linting ran successfully.

Please see the log below for more details:

See the log below:

<details>
<summary>Log</summary>
```
2024-09-18T06:24:13.4603437Z Current runner version: '2.319.1'
2024-09-18T06:24:13.4629356Z ##[group]Operating System
2024-09-18T06:24:13.4630009Z Ubuntu
2024-09-18T06:24:13.4630474Z 22.04.5
2024-09-18T06:24:13.4630775Z LTS
2024-09-18T06:24:13.4631112Z ##[endgroup]
2024-09-18T06:24:13.4631574Z ##[group]Runner Image
2024-09-18T06:24:13.4632167Z Image: ubuntu-22.04
2024-09-18T06:24:13.4632578Z Version: 20240915.1.0
2024-09-18T06:24:13.4633656Z Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240915.1/images/ubuntu/Ubuntu2204-Readme.md
2024-09-18T06:24:13.4635118Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240915.1
2024-09-18T06:24:13.4635942Z ##[endgroup]
2024-09-18T06:24:13.4636420Z ##[group]Runner Image Provisioner
2024-09-18T06:24:13.4636898Z 2.0.384.1
2024-09-18T06:24:13.4637222Z ##[endgroup]
2024-09-18T06:24:13.4652714Z ##[group]GITHUB_TOKEN Permissions
2024-09-18T06:24:13.4654417Z Contents: read
2024-09-18T06:24:13.4654860Z Metadata: read
2024-09-18T06:24:13.4655391Z Packages: read
2024-09-18T06:24:13.4655937Z ##[endgroup]
2024-09-18T06:24:13.4658957Z Secret source: Actions
2024-09-18T06:24:13.4659638Z Prepare workflow directory
2024-09-18T06:24:13.5410432Z Prepare all required actions
2024-09-18T06:24:13.5571769Z Getting action download info
2024-09-18T06:24:13.7577916Z Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
2024-09-18T06:24:13.8642846Z Download action repository 'actions/setup-python@v5' (SHA:f677139bbe7f9c59b41e40162b753c062f5d49a3)
2024-09-18T06:24:14.0889928Z Complete job name: python-linting
2024-09-18T06:24:14.1838748Z ##[group]Run actions/checkout@v4
2024-09-18T06:24:14.1839454Z with:
2024-09-18T06:24:14.1839860Z   repository: apcamargo/pydustmasker
2024-09-18T06:24:14.1840577Z   token: ***
2024-09-18T06:24:14.1841065Z   ssh-strict: true
2024-09-18T06:24:14.1841459Z   ssh-user: git
2024-09-18T06:24:14.1841826Z   persist-credentials: true
2024-09-18T06:24:14.1842621Z   clean: true
2024-09-18T06:24:14.1843030Z   sparse-checkout-cone-mode: true
2024-09-18T06:24:14.1843516Z   fetch-depth: 1
2024-09-18T06:24:14.1843974Z   fetch-tags: false
2024-09-18T06:24:14.1844378Z   show-progress: true
2024-09-18T06:24:14.1844742Z   lfs: false
2024-09-18T06:24:14.1845168Z   submodules: false
2024-09-18T06:24:14.1845571Z   set-safe-directory: true
2024-09-18T06:24:14.1845967Z env:
2024-09-18T06:24:14.1846387Z   CARGO_TERM_COLOR: always
2024-09-18T06:24:14.1846827Z   RUST_BACKTRACE: 1
2024-09-18T06:24:14.1847210Z   RUSTFLAGS: -D warnings
2024-09-18T06:24:14.1847691Z   PYTHON_VERSION: 3.9
2024-09-18T06:24:14.1848189Z ##[endgroup]
2024-09-18T06:24:14.3723397Z Syncing repository: apcamargo/pydustmasker
2024-09-18T06:24:14.3725535Z ##[group]Getting Git version info
2024-09-18T06:24:14.3726471Z Working directory is '/home/runner/work/pydustmasker/pydustmasker'
2024-09-18T06:24:14.3727657Z [command]/usr/bin/git version
2024-09-18T06:24:14.3766115Z git version 2.46.0
2024-09-18T06:24:14.3792531Z ##[endgroup]
2024-09-18T06:24:14.3807655Z Temporarily overriding HOME='/home/runner/work/_temp/55cfee19-4e4a-408c-b628-a746da38b883' before making global git config changes
2024-09-18T06:24:14.3809223Z Adding repository directory to the temporary git global config as a safe directory
2024-09-18T06:24:14.3820512Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/pydustmasker/pydustmasker
2024-09-18T06:24:14.3857688Z Deleting the contents of '/home/runner/work/pydustmasker/pydustmasker'
2024-09-18T06:24:14.3861123Z ##[group]Initializing the repository
2024-09-18T06:24:14.3866056Z [command]/usr/bin/git init /home/runner/work/pydustmasker/pydustmasker
2024-09-18T06:24:14.3928259Z hint: Using 'master' as the name for the initial branch. This default branch name
2024-09-18T06:24:14.3929877Z hint: is subject to change. To configure the initial branch name to use in all
2024-09-18T06:24:14.3931485Z hint: of your new repositories, which will suppress this warning, call:
2024-09-18T06:24:14.3932909Z hint:
2024-09-18T06:24:14.3933924Z hint: 	git config --global init.defaultBranch <name>
2024-09-18T06:24:14.3935104Z hint:
2024-09-18T06:24:14.3935924Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2024-09-18T06:24:14.3937121Z hint: 'development'. The just-created branch can be renamed via this command:
2024-09-18T06:24:14.3937842Z hint:
2024-09-18T06:24:14.3938256Z hint: 	git branch -m <name>
2024-09-18T06:24:14.3939216Z Initialized empty Git repository in /home/runner/work/pydustmasker/pydustmasker/.git/
2024-09-18T06:24:14.3945604Z [command]/usr/bin/git remote add origin https://github.com/apcamargo/pydustmasker
2024-09-18T06:24:14.3977207Z ##[endgroup]
2024-09-18T06:24:14.3978073Z ##[group]Disabling automatic garbage collection
2024-09-18T06:24:14.3981420Z [command]/usr/bin/git config --local gc.auto 0
2024-09-18T06:24:14.4009039Z ##[endgroup]
2024-09-18T06:24:14.4009727Z ##[group]Setting up auth
2024-09-18T06:24:14.4015881Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-09-18T06:24:14.4049959Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-09-18T06:24:14.4355408Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-09-18T06:24:14.4387779Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-09-18T06:24:14.4616280Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2024-09-18T06:24:14.4657514Z ##[endgroup]
2024-09-18T06:24:14.4658787Z ##[group]Fetching the repository
2024-09-18T06:24:14.4668141Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +512e209f5c51ed0b7ca0c73f732b488184b0034f:refs/remotes/origin/main
2024-09-18T06:24:14.9031166Z From https://github.com/apcamargo/pydustmasker
2024-09-18T06:24:14.9032936Z  * [new ref]         512e209f5c51ed0b7ca0c73f732b488184b0034f -> origin/main
2024-09-18T06:24:14.9054916Z ##[endgroup]
2024-09-18T06:24:14.9056294Z ##[group]Determining the checkout info
2024-09-18T06:24:14.9057659Z ##[endgroup]
2024-09-18T06:24:14.9062331Z [command]/usr/bin/git sparse-checkout disable
2024-09-18T06:24:14.9100739Z [command]/usr/bin/git config --local --unset-all extensions.worktreeConfig
2024-09-18T06:24:14.9128284Z ##[group]Checking out the ref
2024-09-18T06:24:14.9131628Z [command]/usr/bin/git checkout --progress --force -B main refs/remotes/origin/main
2024-09-18T06:24:14.9184911Z Switched to a new branch 'main'
2024-09-18T06:24:14.9188284Z branch 'main' set up to track 'origin/main'.
2024-09-18T06:24:14.9194373Z ##[endgroup]
2024-09-18T06:24:14.9231530Z [command]/usr/bin/git log -1 --format='%H'
2024-09-18T06:24:14.9255755Z '512e209f5c51ed0b7ca0c73f732b488184b0034f'
2024-09-18T06:24:14.9587230Z ##[group]Run actions/setup-python@v5
2024-09-18T06:24:14.9587918Z with:
2024-09-18T06:24:14.9588318Z   python-version: 3.9
2024-09-18T06:24:14.9588792Z   cache: pip
2024-09-18T06:24:14.9589219Z   check-latest: false
2024-09-18T06:24:14.9589856Z   token: ***
2024-09-18T06:24:14.9590300Z   update-environment: true
2024-09-18T06:24:14.9590822Z   allow-prereleases: false
2024-09-18T06:24:14.9591285Z env:
2024-09-18T06:24:14.9591667Z   CARGO_TERM_COLOR: always
2024-09-18T06:24:14.9592304Z   RUST_BACKTRACE: 1
2024-09-18T06:24:14.9592775Z   RUSTFLAGS: -D warnings
2024-09-18T06:24:14.9593253Z   PYTHON_VERSION: 3.9
2024-09-18T06:24:14.9593693Z ##[endgroup]
2024-09-18T06:24:15.1193416Z ##[group]Installed versions
2024-09-18T06:24:15.1255928Z Successfully set up CPython (3.9.20)
2024-09-18T06:24:15.1257643Z ##[endgroup]
2024-09-18T06:24:15.2093614Z [command]/opt/hostedtoolcache/Python/3.9.20/x64/bin/pip cache dir
2024-09-18T06:24:16.9952155Z /home/runner/.cache/pip
2024-09-18T06:24:17.3732465Z pip cache is not found
2024-09-18T06:24:17.3867904Z ##[group]Run python -m pip install -U pip
2024-09-18T06:24:17.3868506Z [36;1mpython -m pip install -U pip[0m
2024-09-18T06:24:17.3869034Z [36;1mpython -m pip install -r requirements-dev.txt[0m
2024-09-18T06:24:17.3869625Z [36;1mpython -m pip install -e .[0m
2024-09-18T06:24:17.3870068Z [36;1mmaturin build --out dist[0m
2024-09-18T06:24:17.3870633Z [36;1mpython -m pip install --no-index --find-links=dist/ prelude-parser[0m
2024-09-18T06:24:17.3900530Z shell: /usr/bin/bash -e {0}
2024-09-18T06:24:17.3901033Z env:
2024-09-18T06:24:17.3901348Z   CARGO_TERM_COLOR: always
2024-09-18T06:24:17.3901730Z   RUST_BACKTRACE: 1
2024-09-18T06:24:17.3902435Z   RUSTFLAGS: -D warnings
2024-09-18T06:24:17.3902802Z   PYTHON_VERSION: 3.9
2024-09-18T06:24:17.3903253Z   pythonLocation: /opt/hostedtoolcache/Python/3.9.20/x64
2024-09-18T06:24:17.3903950Z   PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.9.20/x64/lib/pkgconfig
2024-09-18T06:24:17.3904707Z   Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.20/x64
2024-09-18T06:24:17.3905334Z   Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.20/x64
2024-09-18T06:24:17.3905908Z   Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.20/x64
2024-09-18T06:24:17.3906438Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.20/x64/lib
2024-09-18T06:24:17.3906976Z ##[endgroup]
2024-09-18T06:24:18.5790701Z Requirement already satisfied: pip in /opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages (24.2)
2024-09-18T06:24:19.1811065Z Obtaining file:///home/runner/work/pydustmasker/pydustmasker (from -r requirements-dev.txt (line 7))
2024-09-18T06:24:19.1841263Z   Installing build dependencies: started
2024-09-18T06:24:20.0959658Z   Installing build dependencies: finished with status 'done'
2024-09-18T06:24:20.0970254Z   Checking if build backend supports build_editable: started
2024-09-18T06:24:20.1641341Z   Checking if build backend supports build_editable: finished with status 'done'
2024-09-18T06:24:20.1652563Z   Getting requirements to build editable: started
2024-09-18T06:24:20.2119528Z   Getting requirements to build editable: finished with status 'done'
2024-09-18T06:24:20.2132639Z   Preparing editable metadata (pyproject.toml): started
2024-09-18T06:24:22.8757188Z   Preparing editable metadata (pyproject.toml): finished with status 'done'
2024-09-18T06:24:23.0267426Z Collecting mypy==1.11.2 (from -r requirements-dev.txt (line 1))
2024-09-18T06:24:23.0362263Z   Downloading mypy-1.11.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_28_x86_64.whl.metadata (1.9 kB)
2024-09-18T06:24:23.0932767Z Collecting pre-commit==3.8.0 (from -r requirements-dev.txt (line 2))
2024-09-18T06:24:23.0971112Z   Downloading pre_commit-3.8.0-py2.py3-none-any.whl.metadata (1.3 kB)
2024-09-18T06:24:23.1683955Z Collecting pytest==8.3.3 (from -r requirements-dev.txt (line 3))
2024-09-18T06:24:23.1721066Z   Downloading pytest-8.3.3-py3-none-any.whl.metadata (7.5 kB)
2024-09-18T06:24:23.1981478Z Collecting pytest-cov==5.0.0 (from -r requirements-dev.txt (line 4))
2024-09-18T06:24:23.2067813Z   Downloading pytest_cov-5.0.0-py3-none-any.whl.metadata (27 kB)
2024-09-18T06:24:23.6443717Z Collecting ruff==0.6.5 (from -r requirements-dev.txt (line 5))
2024-09-18T06:24:23.6487755Z   Downloading ruff-0.6.5-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (25 kB)
2024-09-18T06:24:23.8177093Z Collecting maturin==1.7.1 (from -r requirements-dev.txt (line 6))
2024-09-18T06:24:23.8198335Z   Using cached maturin-1.7.1-py3-none-manylinux_2_12_x86_64.manylinux2010_x86_64.musllinux_1_1_x86_64.whl.metadata (18 kB)
2024-09-18T06:24:23.8434279Z Collecting typing-extensions>=4.6.0 (from mypy==1.11.2->-r requirements-dev.txt (line 1))
2024-09-18T06:24:23.8473206Z   Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
2024-09-18T06:24:23.8594382Z Collecting mypy-extensions>=1.0.0 (from mypy==1.11.2->-r requirements-dev.txt (line 1))
2024-09-18T06:24:23.8634156Z   Downloading mypy_extensions-1.0.0-py3-none-any.whl.metadata (1.1 kB)
2024-09-18T06:24:23.8800396Z Collecting tomli>=1.1.0 (from mypy==1.11.2->-r requirements-dev.txt (line 1))
2024-09-18T06:24:23.8818229Z   Using cached tomli-2.0.1-py3-none-any.whl.metadata (8.9 kB)
2024-09-18T06:24:23.8975977Z Collecting cfgv>=2.0.0 (from pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:23.9029112Z   Downloading cfgv-3.4.0-py2.py3-none-any.whl.metadata (8.5 kB)
2024-09-18T06:24:23.9829165Z Collecting identify>=1.0.0 (from pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:23.9871320Z   Downloading identify-2.6.1-py2.py3-none-any.whl.metadata (4.4 kB)
2024-09-18T06:24:24.0103535Z Collecting nodeenv>=0.11.1 (from pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:24.0142150Z   Downloading nodeenv-1.9.1-py2.py3-none-any.whl.metadata (21 kB)
2024-09-18T06:24:24.0622591Z Collecting pyyaml>=5.1 (from pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:24.0668233Z   Downloading PyYAML-6.0.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
2024-09-18T06:24:24.1675906Z Collecting virtualenv>=20.10.0 (from pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:24.1716232Z   Downloading virtualenv-20.26.5-py3-none-any.whl.metadata (4.5 kB)
2024-09-18T06:24:24.1903000Z Collecting iniconfig (from pytest==8.3.3->-r requirements-dev.txt (line 3))
2024-09-18T06:24:24.1952839Z   Downloading iniconfig-2.0.0-py3-none-any.whl.metadata (2.6 kB)
2024-09-18T06:24:24.2221626Z Collecting packaging (from pytest==8.3.3->-r requirements-dev.txt (line 3))
2024-09-18T06:24:24.2263624Z   Downloading packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
2024-09-18T06:24:24.2480947Z Collecting pluggy<2,>=1.5 (from pytest==8.3.3->-r requirements-dev.txt (line 3))
2024-09-18T06:24:24.2518853Z   Downloading pluggy-1.5.0-py3-none-any.whl.metadata (4.8 kB)
2024-09-18T06:24:24.2710359Z Collecting exceptiongroup>=1.0.0rc8 (from pytest==8.3.3->-r requirements-dev.txt (line 3))
2024-09-18T06:24:24.2746545Z   Downloading exceptiongroup-1.2.2-py3-none-any.whl.metadata (6.6 kB)
2024-09-18T06:24:24.6519578Z Collecting coverage>=5.2.1 (from coverage[toml]>=5.2.1->pytest-cov==5.0.0->-r requirements-dev.txt (line 4))
2024-09-18T06:24:24.6569313Z   Downloading coverage-7.6.1-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (8.3 kB)
2024-09-18T06:24:24.6847301Z Collecting distlib<1,>=0.3.7 (from virtualenv>=20.10.0->pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:24.6881513Z   Downloading distlib-0.3.8-py2.py3-none-any.whl.metadata (5.1 kB)
2024-09-18T06:24:24.7187763Z Collecting filelock<4,>=3.12.2 (from virtualenv>=20.10.0->pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:24.7229045Z   Downloading filelock-3.16.1-py3-none-any.whl.metadata (2.9 kB)
2024-09-18T06:24:24.7534185Z Collecting platformdirs<5,>=3.9.1 (from virtualenv>=20.10.0->pre-commit==3.8.0->-r requirements-dev.txt (line 2))
2024-09-18T06:24:24.7582411Z   Downloading platformdirs-4.3.6-py3-none-any.whl.metadata (11 kB)
2024-09-18T06:24:24.7763054Z Downloading mypy-1.11.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_28_x86_64.whl (12.5 MB)
2024-09-18T06:24:24.8653119Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.5/12.5 MB 143.4 MB/s eta 0:00:00
2024-09-18T06:24:24.8690107Z Downloading pre_commit-3.8.0-py2.py3-none-any.whl (204 kB)
2024-09-18T06:24:24.8793707Z Downloading pytest-8.3.3-py3-none-any.whl (342 kB)
2024-09-18T06:24:24.8868813Z Downloading pytest_cov-5.0.0-py3-none-any.whl (21 kB)
2024-09-18T06:24:24.8962871Z Downloading ruff-0.6.5-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.3 MB)
2024-09-18T06:24:24.9644029Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.3/10.3 MB 153.6 MB/s eta 0:00:00
2024-09-18T06:24:24.9664028Z Using cached maturin-1.7.1-py3-none-manylinux_2_12_x86_64.manylinux2010_x86_64.musllinux_1_1_x86_64.whl (8.7 MB)
2024-09-18T06:24:24.9771570Z Downloading cfgv-3.4.0-py2.py3-none-any.whl (7.2 kB)
2024-09-18T06:24:24.9927592Z Downloading coverage-7.6.1-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (234 kB)
2024-09-18T06:24:25.0006437Z Downloading exceptiongroup-1.2.2-py3-none-any.whl (16 kB)
2024-09-18T06:24:25.0080099Z Downloading identify-2.6.1-py2.py3-none-any.whl (98 kB)
2024-09-18T06:24:25.0146675Z Downloading mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
2024-09-18T06:24:25.0209743Z Downloading nodeenv-1.9.1-py2.py3-none-any.whl (22 kB)
2024-09-18T06:24:25.0272610Z Downloading pluggy-1.5.0-py3-none-any.whl (20 kB)
2024-09-18T06:24:25.0340858Z Downloading PyYAML-6.0.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (737 kB)
2024-09-18T06:24:25.0423108Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 737.4/737.4 kB 89.1 MB/s eta 0:00:00
2024-09-18T06:24:25.0441527Z Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
2024-09-18T06:24:25.0478679Z Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
2024-09-18T06:24:25.0575323Z Downloading virtualenv-20.26.5-py3-none-any.whl (6.0 MB)
2024-09-18T06:24:25.1067812Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.0/6.0 MB 123.8 MB/s eta 0:00:00
2024-09-18T06:24:25.1107001Z Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
2024-09-18T06:24:25.1174749Z Downloading packaging-24.1-py3-none-any.whl (53 kB)
2024-09-18T06:24:25.1245074Z Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
2024-09-18T06:24:25.1344410Z Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
2024-09-18T06:24:25.1415430Z Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
2024-09-18T06:24:25.1989220Z Building wheels for collected packages: pydustmasker
2024-09-18T06:24:25.1999002Z   Building editable for pydustmasker (pyproject.toml): started
2024-09-18T06:24:38.5767374Z   Building editable for pydustmasker (pyproject.toml): finished with status 'done'
2024-09-18T06:24:38.5775321Z   Created wheel for pydustmasker: filename=pydustmasker-0.1.0-cp39-cp39-linux_x86_64.whl size=2022 sha256=631b73d956599e638080f58643237666fc426978e48dbbce18ac20140813777a
2024-09-18T06:24:38.5777346Z   Stored in directory: /tmp/pip-ephem-wheel-cache-6m9qkbd4/wheels/1e/c1/f5/42b8fcea54a979e233b6ff9943c3b42d868e845f68440a4c24
2024-09-18T06:24:38.5796810Z Successfully built pydustmasker
2024-09-18T06:24:38.6283415Z Installing collected packages: pydustmasker, distlib, typing-extensions, tomli, ruff, pyyaml, pluggy, platformdirs, packaging, nodeenv, mypy-extensions, iniconfig, identify, filelock, exceptiongroup, coverage, cfgv, virtualenv, pytest, mypy, maturin, pytest-cov, pre-commit
2024-09-18T06:24:40.8533382Z Successfully installed cfgv-3.4.0 coverage-7.6.1 distlib-0.3.8 exceptiongroup-1.2.2 filelock-3.16.1 identify-2.6.1 iniconfig-2.0.0 maturin-1.7.1 mypy-1.11.2 mypy-extensions-1.0.0 nodeenv-1.9.1 packaging-24.1 platformdirs-4.3.6 pluggy-1.5.0 pre-commit-3.8.0 pydustmasker-0.1.0 pytest-8.3.3 pytest-cov-5.0.0 pyyaml-6.0.2 ruff-0.6.5 tomli-2.0.1 typing-extensions-4.12.2 virtualenv-20.26.5
2024-09-18T06:24:41.4626886Z Obtaining file:///home/runner/work/pydustmasker/pydustmasker
2024-09-18T06:24:41.4666849Z   Installing build dependencies: started
2024-09-18T06:24:42.3043297Z   Installing build dependencies: finished with status 'done'
2024-09-18T06:24:42.3054071Z   Checking if build backend supports build_editable: started
2024-09-18T06:24:42.3518437Z   Checking if build backend supports build_editable: finished with status 'done'
2024-09-18T06:24:42.3529366Z   Getting requirements to build editable: started
2024-09-18T06:24:42.3988830Z   Getting requirements to build editable: finished with status 'done'
2024-09-18T06:24:42.4000405Z   Preparing editable metadata (pyproject.toml): started
2024-09-18T06:24:42.6393543Z   Preparing editable metadata (pyproject.toml): finished with status 'done'
2024-09-18T06:24:42.6427158Z Building wheels for collected packages: pydustmasker
2024-09-18T06:24:42.6437401Z   Building editable for pydustmasker (pyproject.toml): started
2024-09-18T06:24:42.9118975Z   Building editable for pydustmasker (pyproject.toml): finished with status 'done'
2024-09-18T06:24:42.9127023Z   Created wheel for pydustmasker: filename=pydustmasker-0.1.0-cp39-cp39-linux_x86_64.whl size=2022 sha256=5367522e0566b9ca845f2d18fced5157a957e252ce6083f33e996feaf3900543
2024-09-18T06:24:42.9129133Z   Stored in directory: /tmp/pip-ephem-wheel-cache-7xh4ki9k/wheels/1e/c1/f5/42b8fcea54a979e233b6ff9943c3b42d868e845f68440a4c24
2024-09-18T06:24:42.9152730Z Successfully built pydustmasker
2024-09-18T06:24:42.9437893Z Installing collected packages: pydustmasker
2024-09-18T06:24:42.9439177Z   Attempting uninstall: pydustmasker
2024-09-18T06:24:42.9445295Z     Found existing installation: pydustmasker 0.1.0
2024-09-18T06:24:42.9465875Z     Uninstalling pydustmasker-0.1.0:
2024-09-18T06:24:42.9645103Z       Successfully uninstalled pydustmasker-0.1.0
2024-09-18T06:24:42.9766113Z Successfully installed pydustmasker-0.1.0
2024-09-18T06:24:43.0849894Z 📦 Including license file "/home/runner/work/pydustmasker/pydustmasker/LICENSE"
2024-09-18T06:24:43.0851590Z 🍹 Building a mixed python/rust project
2024-09-18T06:24:43.0852895Z 🔗 Found pyo3 bindings
2024-09-18T06:24:43.1605244Z 🐍 Found CPython 3.9 at /opt/hostedtoolcache/Python/3.9.20/x64/bin/python3
2024-09-18T06:24:43.1606840Z 📡 Using build options features from pyproject.toml
2024-09-18T06:24:43.2195589Z [1m[32m   Compiling[0m target-lexicon v0.12.16
2024-09-18T06:24:43.2196508Z [1m[32m   Compiling[0m once_cell v1.19.0
2024-09-18T06:24:43.2197416Z [1m[32m   Compiling[0m proc-macro2 v1.0.86
2024-09-18T06:24:43.2204922Z [1m[32m   Compiling[0m unicode-ident v1.0.13
2024-09-18T06:24:43.2815508Z [1m[32m   Compiling[0m autocfg v1.3.0
2024-09-18T06:24:43.3831412Z [1m[32m   Compiling[0m libc v0.2.158
2024-09-18T06:24:43.5994715Z [1m[32m   Compiling[0m memoffset v0.9.1
2024-09-18T06:24:43.7211438Z [1m[32m   Compiling[0m heck v0.5.0
2024-09-18T06:24:43.9184033Z [1m[32m   Compiling[0m unindent v0.2.3
2024-09-18T06:24:43.9467364Z [1m[32m   Compiling[0m pyo3-build-config v0.22.3
2024-09-18T06:24:44.0474136Z [1m[32m   Compiling[0m quote v1.0.37
2024-09-18T06:24:44.2800100Z [1m[32m   Compiling[0m syn v2.0.77
2024-09-18T06:24:44.2965382Z [1m[32m   Compiling[0m indoc v2.0.5
2024-09-18T06:24:44.4419483Z [1m[32m   Compiling[0m cfg-if v1.0.0
2024-09-18T06:24:45.1444106Z [1m[32m   Compiling[0m pyo3-macros-backend v0.22.3
2024-09-18T06:24:45.1445326Z [1m[32m   Compiling[0m pyo3-ffi v0.22.3
2024-09-18T06:24:45.1447340Z [1m[32m   Compiling[0m pyo3 v0.22.3
2024-09-18T06:24:48.6507540Z [1m[32m   Compiling[0m pyo3-macros v0.22.3
2024-09-18T06:24:51.0468378Z [1m[32m   Compiling[0m pydustmasker v0.1.0 (/home/runner/work/pydustmasker/pydustmasker)
2024-09-18T06:24:51.7593000Z [1m[32m    Finished[0m `dev` profile [unoptimized + debuginfo] target(s) in 8.56s
2024-09-18T06:24:52.5715440Z 📦 Built wheel for CPython 3.9 to dist/pydustmasker-0.1.0-cp39-cp39-manylinux_2_34_x86_64.whl
2024-09-18T06:24:52.9672194Z Looking in links: dist/
2024-09-18T06:24:52.9765624Z ERROR: Could not find a version that satisfies the requirement prelude-parser (from versions: none)
2024-09-18T06:24:52.9767752Z ERROR: No matching distribution found for prelude-parser
2024-09-18T06:24:53.0215088Z ##[error]Process completed with exit code 1.
2024-09-18T06:24:53.0322557Z Post job cleanup.
2024-09-18T06:24:53.1269403Z [command]/usr/bin/git version
2024-09-18T06:24:53.1307090Z git version 2.46.0
2024-09-18T06:24:53.1356280Z Temporarily overriding HOME='/home/runner/work/_temp/de5bfad5-43b6-4b75-b31c-091e03756b73' before making global git config changes
2024-09-18T06:24:53.1357401Z Adding repository directory to the temporary git global config as a safe directory
2024-09-18T06:24:53.1369900Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/pydustmasker/pydustmasker
2024-09-18T06:24:53.1406585Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-09-18T06:24:53.1438680Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-09-18T06:24:53.1686399Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-09-18T06:24:53.1714400Z http.https://github.com/.extraheader
2024-09-18T06:24:53.1727449Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-09-18T06:24:53.1762084Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-09-18T06:24:53.2232588Z Cleaning up orphan processes
```
</details>

**Comment by sanders41** [comment]
Thanks for reporting this, it is a bug. I had it in there while testing and forgot to remove it. Would you like to do a PR to remove it? It is in the `github_actions.rs` file.

**Comment by apcamargo** [comment]
Sure!

**Comment by apcamargo** [comment]
Done!

I also noticed that the CI is using some deprecated actions that will need to be replaced in the future, such as `release-drafter@v5`, `rust-cache@v2.6.2`, and `upload-artifact@v3`. Since I haven't encountered any issues with them so far, I left them as they are.

Thank you for this tool, it is incredibly useful. It's unfortunate that it took me so long to discover it. Do you plan to write documentation? There are a few features I haven't used yet and would like to learn more about, such as the release drafter and the benefits of using the justfile over a traditional makefile.

**Comment by sanders41** [comment]
Yes in testing your PR I also noticed there are some actions that need updating. You are welcome to update them if you like, otherwise I will do it.

Do you have thoughts around what needs additional documentation? I'm happy to accept PRs for that also, or to try to better document things you are unsure about.

The main benefits in my opinion to just are it woks on Windows also (make doesn't) and it was built with the purpose of running commands like this. Since make was really built for build C projects you sometimes need tricks like `phony`, just doesn't need this.

**Comment by sanders41** [comment]
Got it, yes I can do that. I created a tracking issue here #464

**Comment by sanders41** [comment]
Closed via #458
